analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ParametersCorrectOrder.cs

```diff
@@ -193,4 +193,41 @@ public void M3()
             double doTheThing(int divisor, int dividend) => divide(dividend, divisor);  // Noncompliant
         }
     }
+
+    // See https://github.com/SonarSource/sonar-dotnet/issues/3879
+    class NullableParam
+    {
+        void NotNullableParamVoid(int a, int b) // Secondary [D,E,F]
+        {
+            // Do nothing
+        }
+
+        void NullableParamValueVoid(int a, int? b)
+        {
+            if (b.HasValue)
+            {
+                NotNullableParamVoid(b.Value, a); // Noncompliant [D] FN
+                NotNullableParamVoid(a, b.Value); // Compliant [D]
```

Comment:
Flow is not tracked for Compliant cases. It's just fine as it is.
```suggestion
                NotNullableParamVoid(a, b.Value); // Compliant
```

Reply:
Same for cases below